File: src/test/java/test/java/CryptoTest.java
package test.java;

import me.saro.dat.DatUtils;
import me.saro.dat.crypto.DatCrypto;
import me.saro.dat.crypto.DatCryptoAlgorithm;
import org.junit.jupiter.api.Test;

import java.util.Arrays;

import static org.junit.jupiter.api.Assertions.assertThrows;

public class CryptoTest {

    public void unit(DatCryptoAlgorithm alg) {
        String tag = "Crypto " + alg;

        byte[] body = DatUtils.generateRandomBase62(100).getBytes();
        DatCrypto cryptoKey = DatCrypto.generate(alg);
        DatCrypto cryptoKeyFail = DatCrypto.generate(alg);
        byte[] cryptoKeyBytes = cryptoKey.toBytes();
        DatCrypto cryptoKeyFrom = DatCrypto.fromBytes(alg, cryptoKeyBytes);

        var encrypted = cryptoKeyFrom.encrypt(body);

        assert Arrays.equals(cryptoKey.decrypt(encrypted), body);
        assert Arrays.equals(cryptoKeyFrom.decrypt(encrypted), body);
        assertThrows(Exception.class, () -> cryptoKeyFail.decrypt(encrypted));
        System.out.println(tag + " PASS : " + DatUtils.encodeBase64Url(cryptoKeyBytes));
    }

    @Test
    public void test() {
        for (var algorithm : DatCryptoAlgorithm.getEntries()) {
            for (var i = 0; i < 20; i++) {
                unit(algorithm);
            }
        }
    }
}

File: src/test/kotlin/test/kt/CryptoTest.kt
package test.kt

import me.saro.dat.DatUtils.Companion.encodeBase64Url
import me.saro.dat.DatUtils.Companion.generateRandomBase62
import me.saro.dat.crypto.DatCrypto.Companion.fromBytes
import me.saro.dat.crypto.DatCrypto.Companion.generate
import me.saro.dat.crypto.DatCryptoAlgorithm
import org.junit.jupiter.api.Assertions
import org.junit.jupiter.api.Test
import org.junit.jupiter.api.function.Executable

class CryptoTest {
    fun unit(alg: DatCryptoAlgorithm) {
        val tag = "Crypto " + alg

        val body = generateRandomBase62(100).toByteArray()
        val cryptoKey = generate(alg)
        val cryptoKeyFail = generate(alg)
        val cryptoKeyBytes = cryptoKey.toBytes()
        val cryptoKeyFrom = fromBytes(alg, cryptoKeyBytes)

        val encrypted = cryptoKeyFrom.encrypt(body)

        assert(cryptoKey.decrypt(encrypted).contentEquals(body))
        assert(cryptoKeyFrom.decrypt(encrypted).contentEquals(body))
        Assertions.assertThrows(Exception::class.java, Executable { cryptoKeyFail.decrypt(encrypted) })
        println(tag + " PASS : " + encodeBase64Url(cryptoKeyBytes))
    }

    @Test
    fun test() {
        for (algorithm in DatCryptoAlgorithm.entries) {
            for (i in 0..19) {
                unit(algorithm)
            }
        }
    }
}
